src/rules/no-deprecated-functions.ts

```diff
@@ -1,9 +1,56 @@
+import { JSONSchemaForNPMPackageJsonFiles } from '@schemastore/package';
 import {
   AST_NODE_TYPES,
   TSESTree,
 } from '@typescript-eslint/experimental-utils';
 import { createRule, getNodeName } from './utils';
 
+interface ContextSettings {
+  jest?: EslintPluginJestSettings;
+}
+
+export type JestVersion =
+  | 14
+  | 15
+  | 16
+  | 17
+  | 18
+  | 19
+  | 20
+  | 21
+  | 22
+  | 23
+  | 24
+  | 25
+  | 26
+  | 27
+  | number;
+
+interface EslintPluginJestSettings {
+  version: JestVersion;
+}
+
+const detectJestVersion = (): JestVersion => {
```

Comment:
could we make this so that once we've found the version it's cached? so we don't do extra fs per test file.

For the test you might need `jest.resetModules()` so it'll be recalculated

Reply:
For now I've just gone with a one line `_clearCachedJestVersion` function, as `jest.resetModules` didn't work, and so we'd have to have that even if it was in another file.

I'll break it out into another file later :)

Reply:
you'd need to re-import the file after clearing so you don't hold on to a reference to the old module.

no biggie, tho